qlobber-sub.js

```diff
@@ -23,14 +23,26 @@ QlobberSub.prototype._add_value = function (clientMap, val) {
   topicMap.set(val.topic, val.qos)
 }
 
-QlobberSub.prototype._add_values = function (dest, clientMap) {
+QlobberSub.prototype._add_values = function (dest, clientMap, topic) {
   for (var clientIdAndTopicMap of clientMap) {
-    for (var topicAndQos of clientIdAndTopicMap[1]) {
-      dest.push({
-        clientId: clientIdAndTopicMap[0],
-        topic: topicAndQos[0],
-        qos: topicAndQos[1]
-      })
+    var clientId = clientIdAndTopicMap[0]
+    var topicMap = clientIdAndTopicMap[1]
+    if (topic === undefined) {
+      for (var topicAndQos of topicMap) {
+        dest.push({
+          clientId: clientId,
+          topic: topicAndQos[0],
+          qos: topicAndQos[1]
+        })
+      }
```

Comment:
Aside: In a simple test I wrote, adding 10 million strings to a `Map`:

```javascript
var map = new Map();

for (var i = 0; i < 10000000; i += 1)
{
    map.set(i + 'hello', i + 'there');
} 
```

is way faster than adding to an Object

```javascript
var obj = {};

for (var i = 0; i < 10000000; i += 1)
{
    obj[i + 'hello'] = i + 'there';
}
```